### PR TITLE
[api][ci] Move creation of user with fake pw to model

### PR DESCRIPTION
**Pull request by eduardoj**
Add basic tests for the new method.

Pair programming with @bgeuken

### CODE REVIEW

**Comment by bgeuken** [comment]
@openSUSE/open-build-service anyone mind to review?

File: src/api/spec/models/user_spec.rb [hunk 1]
```diff
@@ -267,4 +267,22 @@
...
   end
+
+  describe 'create_user_with_fake_pw!' do
+    context 'with login and email' do
+      let(:user) { User.create_user_with_fake_pw!({ login: 'tux', email: 'some@email.com' }) }
+
+      it 'creates a user with a fake password' do
+        expect(user).to be_an(User)
+        expect(user.login).to eq('tux')
+        expect(user.email).to eq('some@email.com')
```
Comment (ChrisBr) [inline src/api/spec/models/user_spec.rb]: It is not necessary to check the pw as it is anyway validated, right?

Comment (bgeuken) [inline src/api/spec/models/user_spec.rb]: right. And I wouldn't know how to test that it's truly random^^

Comment (bgeuken) [inline src/api/spec/models/user_spec.rb]: Well, creating two users and compare the pw would work

Comment (bgeuken) [inline src/api/spec/models/user_spec.rb]: Just discussed that with Christian. I'll add a simple check in another PR

**Pull request merged**